### PR TITLE
NEW:Invoice creation from task Time

### PR DESCRIPTION
**Pull request by delcroip**
NEW:Invoice creation from task Time
no link created yet,
trad needed

### CODE REVIEW

**Comment by simnandez** [comment]
@delcroip I can not find a issue type feature request about this (It is preferable to first create a issue for To explain the idea, discuss it ...), Can you explain the purpose and functionality of the idea?

**Comment by delcroip** [comment]
Hello,
The idea is to be able to create an invoice based on the task time logged
for one project. In the first step, one have to select the project, month,
year, customers, grouping méthod (by task, user, task and user)  and task
to invoice(exclude the already invoiced and/or the not approved ),in the
second step, one should link the group task to a service (existing or new)
and can adapt the duration. The third step create the draft invoice with
the correct lines.


Le 16 avr. 2017 11:15, "Juanjo Menent" <notifications@github.com> a écrit :

@delcroip <https://github.com/delcroip> I can not find a issue type feature
request about this (It is preferable to first create a issue for To explain
the idea, discuss it ...), Can you explain the purpose and functionality of
the idea?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/Dolibarr/dolibarr/pull/6662#issuecomment-294341902>,
or mute
the thread

File: htdocs/install/mysql/migration/5.0.0-6.0.0.sql [hunk 1]
```diff
@@ -27,6 +27,7 @@
 ALTER TABLE llx_extrafields ADD COLUMN langs varchar(24);
 
 ALTER TABLE llx_supplier_proposaldet ADD COLUMN fk_unit integer DEFAULT NULL;
+ALTER TABLE llx_projet_task_time ADD COLUMN status  enum('PLANNED','DRAFT','SUBMITTED','APPROVED','CANCELLED','REJECTED','CHALLENGED','INVOICED','UNDERAPPROVAL') DEFAULT 'DRAFT';
```
Comment (eldy) [inline htdocs/install/mysql/migration/5.0.0-6.0.0.sql:30]: Field with type "enum" is not allowed by dolibar sql rules :
https://wiki.dolibarr.org/index.php/Language_and_development_rules#Table_and_fields_structures.
You can set this to varchar and add the enum arrays as a constant into class Tasks.

Comment (delcroip) [reply htdocs/install/mysql/migration/5.0.0-6.0.0.sql:30]: Thank you but I think I'll come back with a solution based on int.  

In the page you mentionned there is no tinyint or Time  format so I assumed there are also forbidden, aren't they ?

br

Comment (eldy) [reply htdocs/install/mysql/migration/5.0.0-6.0.0.sql:30]: tinyint can still be used.
For date, only date and datetime are allowed.

**Comment by delcroip** [comment]
Available in timesheet module

**Pull request closed without merging**